pilot/pkg/model/push_context.go

```diff
@@ -356,6 +357,10 @@ func (first *PushRequest) Merge(other *PushRequest) *PushRequest {
 		Reason: reason,
 	}
 
+	if first.Push != nil && first.Push.Cleanup != nil {
+		first.Push.Cleanup()
```

Comment:
I am not sure I understand the motivation here but I don't think this is right. 99% of the time first.Push and other.Push is the same thing. This is not a pushcontext falling out of scope.

Reply:
Can you recommend a different way to detect pushcontexts falling out of scope?  Should I check to see that first and other have different Ledger Versions?